src/app/pages/sharing/smb/smb-form/smb-form.component.ts

```diff
@@ -152,7 +153,10 @@ export class SmbFormComponent implements OnInit {
 
   form = this.formBuilder.group({
     path: ['', Validators.required],
-    name: ['', [Validators.required]],
+    name: ['', {
+      validators: [Validators.required],
+      asyncValidators: [this.smbNameValidator.bind(this)],
```

Comment:
Noticed that you've used `.bind` , however it's not widely used across the project (only 23 usages of `.bind` in all code files).

I'd suggested to use arrow function `() => ` instead.